editor/test/resources/save_data_project/checked.material

```diff
@@ -159,78 +158,196 @@ attributes {
     v: 1.0
     v: 1.0
   }
+  vector_type: VECTOR_TYPE_VEC3
 }
 attributes {
   name: "normalized_float"
   semantic_type: SEMANTIC_TYPE_COLOR
-  element_count: 3
   normalize: true
   coordinate_space: COORDINATE_SPACE_WORLD
   double_values {
     v: 1.0
     v: 1.0
     v: 1.0
   }
+  vector_type: VECTOR_TYPE_VEC3
 }
 attributes {
   name: "byte"
-  element_count: 1
   data_type: TYPE_BYTE
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "unsigned_byte"
-  element_count: 1
   data_type: TYPE_UNSIGNED_BYTE
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "short"
-  element_count: 1
   data_type: TYPE_SHORT
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "unsigned_short"
-  element_count: 1
   data_type: TYPE_UNSIGNED_SHORT
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "int"
-  element_count: 1
   data_type: TYPE_INT
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "unsigned_int"
-  element_count: 1
   data_type: TYPE_UNSIGNED_INT
   coordinate_space: COORDINATE_SPACE_WORLD
   long_values {
     v: 1
   }
+  vector_type: VECTOR_TYPE_SCALAR
 }
 attributes {
   name: "float"
-  element_count: 1
   coordinate_space: COORDINATE_SPACE_WORLD
   double_values {
     v: 1.0
   }
+  vector_type: VECTOR_TYPE_SCALAR
+}
+attributes {
+  name: "instance_world"
+  semantic_type: SEMANTIC_TYPE_WORLD_MATRIX
+  double_values {
+    v: 1.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 1.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 1.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 0.0
+    v: 1.0
+  }
```

Comment:
Maybe we shouldn't write attributes to the project files when they have a semantic type that will be supplied by the engine / editor?

I guess we could strip this out later if we wanted to. Just feels a bit excessive for these new matrix attributes.

Reply:
Righ, yes that makes sense 👍 same goes for all vector types I suppose